### ISSUE TITLE
Create apis that expose the following User stats

### ISSUE THREAD

**Issue by sheenarbw** [Umuzi-org/Tilde#157]
add a "stats" action to the user viewset. Expose the following info:

total number of cards complete, in review, etc (each card status)
total number of Reviews done on Tilde
total number of PR reviews done

total number of reviews done in last 7 days 
total nimber of PR reviews done in last 7 days
number of cards completed in the last 7 days
number of cards started in last 7 days

If you think of other user-specific stats that should be drawn on a dashboard, please feel free to make suggestions

**Comment by gilbert-umuzi** [comment]
I suggest we use the current weekly recruit report as a starting point for which fields we should include:

Total projects in your curriculum
Complete projects
Recent momentum (shows how your complete projects have changed week by week over the last month. The last number is your most recent number of complete projects.)
Remaining projects
Reviews done last week
Git PR reviews done last week

Learning time used up (weeks past since start date / total weeks)

Progress estimate (Sam has a script that calculates this)
Recent momentum

(There are 100 progress points in your curriculum. You make progress by completing projects. Projects are weighted by how long they take to complete. Generally the later projects are harder, take longer, and are worth more points.)

Tilde project summary (an estimate of whether you will finish all the projects in your curriculum in the time remaining. Sam has a script that calculates the Progress Estimate based weighted projects completed. If you are "Ahead" or "On track" your Progress estimate > Learning time used up. If you are "Falling behind" or "At risk", you aren't completing projects quickly enough and need to put in more time to complete projects and catch up.)

Progress rate
Required rate

(Your Progress rate shows how many progress points you've completed per week since you started. Your Required rate shows the minimum you need to complete per week to finish in the time remaining. )

Total number of cards you've completed
Cards remaining

**Comment by sheenarbw** [comment]
Some of this stuff is trickier than other stuff. Eg momentum would be best achieved by first implementing an Action log model. Which is something I want to do personally because it's tricky.  Weeks remaining means that the backend needs to know about when people started their courses.  

@gilbert-umuzi I'll take your comment and turn it into a bunch of issues.

@EckardBerry  please work according to the original project spec for now. Let's get those quick wins down then add the other stuff afterwards

**Comment by sheenarbw** [comment]
@gilbert-umuzi I made a new story: https://github.com/Umuzi-org/Tilde/issues/186 I'll add your stuff as sub issues so that they can be crunched through. Some of these things need a bit of groundwork to be laid before we can make them happen. Data architecture things need to be set up
